x-pack/plugins/monitoring/server/telemetry_collection/get_all_stats.ts

```diff
@@ -68,14 +69,22 @@ export function handleAllStats(
   }
 ) {
   return clusters.map((cluster) => {
+    const clusterUuid: string =
+      cluster.cluster_uuid !== undefined ? cluster.cluster_uuid : cluster.elasticsearch?.cluster.id;
+
+    const clusterStats =
+      cluster.cluster_stats !== undefined
+        ? cluster.cluster_stats
+        : cluster.elasticsearch?.cluster.stats.stack;
+
     const stats = {
       ...cluster,
```

Comment:
Ah, the confusion here, in the `telemetry_collection_manager ` we are seeking `cluster_uuid` in the `cluster_stats` but here we are using `stack_stats` (in `handleAllStats` of [`get_all_stats.ts`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/server/telemetry_collection/get_all_stats.ts#L107-L115)) to collect Logstash (kibana & beats as well) stats. I think to make this happen, as you mentioned (in some of your comment), we need a normalization (and remove newly added `elasticsearch?:...` structure) to reflect in [`BasicStatsPayload`](https://github.com/elastic/kibana/blob/main/src/plugins/telemetry_collection_manager/server/types.ts#L78).

So, `cluster_uuid` adding wouldn't make any affect in `telemetry_collection_manager`. But this is my understanding while investigation, let me know if I am missing anything since you know better the source.